build-aux/snap/snapcraft.yaml

```diff
@@ -29,173 +45,705 @@ license: GPL-3.0
+      - RELEASE: "22.04.2"
+      - BASE: ubuntu-base-${RELEASE}-base-${CRAFT_TARGET_ARCH}.tar.gz
+      - DIR_URL: https://cdimage.ubuntu.com/ubuntu-base/releases/${RELEASE}/release
+      - URL: ${DIR_URL}/${BASE}
+      - SHA256: ${DIR_URL}/SHA256SUMS
+      - SIG: ${SHA256}.gpg
+      - DEPENDENCIES: >-
+          make
+          gcc
+          binutils
+          gawk
+          python3
+          python3-jinja2
+          meson
+          bison
+          flex
+          ninja-build
+          libc6-dev
+          gperf
+          autoconf
+          automake
+          gettext
+          libtool
+          pkg-config
+          libglib2.0-dev
+          g++
+          autopoint
+          git
+    override-pull: |
+      craftctl default
+      case "${CRAFT_TARGET_ARCH}" in
+        i386)
+        ;;
+        *)
+          wget "${SHA256}" "${SIG}" "${URL}"
+          gpg --no-default-keyring \
+              --keyring ./cd-image-keying.gpg \
+              --verify SHA256SUMS.gpg SHA256SUMS
+          awk -v "file=${BASE}" '$2=="*"file' SHA256SUMS | sha256sum -c
+        ;;
+      esac
+    override-build: |
+      mkdir -p "${CRAFT_PART_INSTALL}/build-base"
+      case "${CRAFT_TARGET_ARCH}" in
+        i386)
+          fakeroot debootstrap --arch=i386 --include "$(echo ${DEPENDENCIES} | sed 's/ /,/g')" bullseye "${CRAFT_PART_INSTALL}/build-base"
+          ;;
+        *)
+          tar -x --xattrs-include=* -f "${BASE}" -C "${CRAFT_PART_INSTALL}/build-base"
+          mkdir -p "${CRAFT_PART_INSTALL}/build-base/snap/snapd/current"
+          "${CRAFT_PROJECT_DIR}/build-aux/snap/local/mount-ns.sh" spawn "${CRAFT_PART_INSTALL}/build-base" -- bash <<EOF
+      set -eu
+      apt update
+      DEBIAN_FRONTEND=noninteractive apt install -y --no-install-recommends ${DEPENDENCIES}
+      EOF
+        ;;
+      esac
+    override-stage: |
+      # Default stage patches files
+      rm -rf "${CRAFT_STAGE}/build-base"
+      cp -raT "${CRAFT_PART_INSTALL}/build-base" "${CRAFT_STAGE}/build-base"
+    override-prime: |
+      # Do nothing
+
+  linux-headers:
+    after:
+     - build-base
+    plugin: nil
+    build-packages:
+      - rsync
+    source-type: git
+    source:  https://git.launchpad.net/~ubuntu-kernel/ubuntu/+source/linux/+git/jammy
+    source-branch: master
+    source-depth: 1
+    override-build: |
+      make INSTALL_HDR_PATH="${CRAFT_PART_INSTALL}" headers_install
+    override-stage: |
+      mkdir -p "${CRAFT_STAGE}/build-base/snap/snapd/current"
+      cp -raT "${CRAFT_PART_INSTALL}" "${CRAFT_STAGE}/build-base/snap/snapd/current"
+    override-prime: |
+      # Do nothing
+
+  glibc:
+    plugin: nil
+    stage-packages:
+      # This is to trigger manifest generation. Files will not end up in snap
+      - base-files
+    after:
+      - linux-headers
+      - build-base
+    build-packages:
+      - attr
+    source-type: git
+    source:  https://git.launchpad.net/ubuntu/+source/glibc
+    source-branch: ubuntu/jammy
```

Comment:
In practice the git repositories did not contain yet the same version as the package on apt repository. It would be a problem if we cannot build now and then.

Reply:
Also there is no `applied/ubuntu/jammy*` on glibc. We will have to apply the patches.